src/main/java/teammates/ui/webapi/OriginCheckFilter.java

```diff
@@ -71,6 +72,16 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             return;
         }
 
+        // The header X-AppEngine-QueueName cannot be spoofed as GAE will strip any user-sent X-AppEngine-QueueName headers.
+        // Reference: https://cloud.google.com/appengine/docs/standard/java/taskqueue/push/creating-handlers#reading_request_headers
+        boolean isRequestFromAppEngineQueue = request.getHeader("X-AppEngine-QueueName") != null;
+
+        if (isRequestFromAppEngineQueue) {
```

Comment:
curious on why you added this logic here, I would have thought this was sufficient:
```
    <filter-mapping>
        <filter-name>OriginCheckFilter</filter-name>
        <url-pattern>/webapi/*</url-pattern>
    </filter-mapping
```

Since the `filter` will never be called for those `worker/` or `auto/` App Engine Queue tasks we have.

Reply:
> Since the filter will never be called for those worker/ or auto/ App Engine Queue tasks we have.

No longer true after the change introduced in this PR.

As for why the origin check filter needs to be extended to them, it's because our APIs are wide open to public (i.e. not restricted by firewalls/IPs) and we need to grant the same security as we do to the more normal (for the lack of better word) APIs in `webapi/`.